ui/src/templates/api/index.ts

```diff
@@ -46,49 +47,61 @@ export const createDashboardFromTemplate = async (
     throw new Error('Failed to create dashboard from template')
   }
 
+  // associate imported label id with new label
+  const labelMap = await createLabelsFromTemplate(template, orgID)
+
   await Promise.all([
-    await createDashboardLabelsFromTemplate(template, createdDashboard),
+    await addDashboardLabelsFromTemplate(template, labelMap, createdDashboard),
     await createCellsFromTemplate(template, createdDashboard),
   ])
-  createVariablesFromTemplate(template, orgID)
+
+  await createVariablesFromTemplate(template, labelMap, orgID)
 
   const dashboard = await client.dashboards.get(createdDashboard.id)
   return dashboard
 }
 
-const createDashboardLabelsFromTemplate = async (
+const addDashboardLabelsFromTemplate = async (
   template: DashboardTemplate,
+  labelMap: LabelMap,
   dashboard: IDashboard
 ) => {
-  const templateLabels = await createLabelsFromTemplate(
-    template,
-    dashboard.orgID
-  )
-  await client.dashboards.addLabels(dashboard.id, templateLabels)
+  const labelRelationships = getLabelRelationships(template.content.data)
+  const labelIDs = labelRelationships.map(l => labelMap[l.id] || '')
+
+  await client.dashboards.addLabels(dashboard.id, labelIDs)
 }
 
+type LabelMap = {[importedID: string]: CreatedLabelID}
+type CreatedLabelID = string
+
 const createLabelsFromTemplate = async <T extends TemplateBase>(
   template: T,
   orgID: string
-) => {
+): Promise<LabelMap> => {
   const {
     content: {data, included},
   } = template
 
-  if (!data.relationships || !data.relationships[TemplateType.Label]) {
-    return
+  const labeledResources = [data, ...included].filter(r =>
+    hasLabelsRelationships(r)
+  )
+
+  if (_.isEmpty(labeledResources)) {
+    return {}
```

Comment:
could this be just a return

Reply:
I wanted to maintain the return type of `LabelMap`. If it's empty the other helpers won't utilize it but it avoids having to add guards for `undefined`.